### ISSUE TITLE
Delayed error responses not handled correctly.

### ISSUE THREAD

**Issue by bencart** [Access4Learning/sif3-framework-java#38]
When we receive an error response from a delayed consumer call we are always getting a 500 error : An error occurred unmarshalling object from XML: java.lang.NullPointerException.

I believe this is because in RemoteMessageQueueReader.processError the ErrorInfo object is built by calling setResponseBaseData(errorInfo, response, MessageType.ERROR);

setResponseBaseData populates the payload of the errorInfo object from response.getDataObject when in reality it should be populated from response.getError

Our response looks like this :
```
Response [dataObjectType=class java.lang.String, dataObject=null, resourceURI=<url>, toString()=BaseResponse [status=200, statusMessage=OK, hasEntity=true, error=ErrorDetails [description=The provider response exceeded the timeout of 60 seconds., errorCode=504, message=<error>, scope=PROVIDER], hdrProperties=HeaderProperties [hdrProperties={date=Tue, 10 Jul 2018 03:59:59 GMT, responsesource=PROVIDER, server=Apache-Coyote/1.1, content-length=351, servicetype=OBJECT, messagetype=ERROR, vary=Accept-Encoding, requestid=<id>, messageid=<id>, relativeservicepath=<servicepath>, connection=keep-alive, content-type=application/xml;charset=UTF-8}], mediaType=application/xml;charset=UTF-8, contentLength=351, zone=null, context=null, delayedReceipt=null]]
```

As you can see dataObject is null. I am not sure if this is pointing to an error further up in the framework (perhaps BaseClient.setResponse) where the error payload is not being set as the dataObject or if this is the intended behaviour of BaseClient.setResponse when an error is received.

**Comment by joerghuber** [comment]
In case of an error the 'error' propery of the 'response' object is populated and not the 'dataObject. As far as this is concerned all looks fine. What I see though is that the HTTP Status returned is 200 (ok) which is wrong. 200 indicates all fine. In that case the framework assumes the data is fine and will attempt to unmarshal it. Now it is an error message rather than a data model object which causes the error. I'm not sure what is best done. For one the HTTP status returned is wrong but then the message type appears to be ERROR. What should the code believe.
I have to check what the framework does and what takes precedence.

**Comment by bencart** [comment]
I will grab the headers next time we have the error. I can reproduce easily but probably not until friday.

The message is detected fine as an error. And is being passed through the framework corrrectly. If it is correct that error property be populated and dataObject be null then the issue is deifinitely  caused in :

RemoteMessageQueueReader.processError
```
ErrorInfo errorInfo = new ErrorInfo();
setResponseBaseData(errorInfo, response, MessageType.ERROR);
```

RemoteMessageQueueReader. setResponseBaseData
```
baseInfo.setPayload((String)response.getDataObject());
```

The payload needs to be set to the error at some point this can't currently happen.

**Comment by bencart** [comment]
Oh sorry Joerg just to clarify the response status 200 is the status code on the request to get the message from the queue. 200 is expected in this case. 

I have emailed you the section of log for one message.

**Comment by joerghuber** [comment]
Found it. When a response is received from the message client the error is already unmarshalled. There is no further payload string. The local message consumer attempted to unmarshal the payload into an error object but this is already done. Fixed now in v0.13.0 on nsip fork. So in case of an error nothing must be done except passing the error object to the onError() method of consumer.

**Comment by joerghuber** [comment]
This is now merged to the master.
